OpenRA.Game/Support/Log.cs

```diff
@@ -75,22 +77,33 @@ public static void AddChannel(string channelName, string baseFilename)
 			}
 		}
 
-		public static void Write(string channel, string value)
+		public static void Write(string channelName, string value)
 		{
-			var writer = Channel(channel).Writer;
+			var channel = Channels[channelName];
+			var writer = channel.Writer;
 			if (writer == null)
 				return;
 
-			writer.WriteLine(value);
+			if (!channel.IsTimestamped)
+				writer.WriteLine(value);
+			else
+				writer.WriteLine("[{0}] {1}", DateTime.Now.ToString(Game.Settings.Server.TimestampFormat), value);
 		}
 
-		public static void Write(string channel, string format, params object[] args)
+		public static void Write(string channelName, string format, params object[] args)
 		{
-			var writer = Channel(channel).Writer;
-			if (writer == null)
+			var channel = Channels[channelName];
+			if (channel.Writer == null)
 				return;
 
-			writer.WriteLine(format, args);
+			if (!channel.IsTimestamped)
+				channel.Writer.WriteLine(format, args);
+			else
+			{
+				var toWrite = string.Format(format, args);
+				toWrite = string.Format("[{0}] {1}", DateTime.Now.ToString(Game.Settings.Server.TimestampFormat), toWrite);
+				channel.Writer.WriteLine(toWrite);
```

Comment:
Please write this as
```csharp
else
{
	var timestamp = DateTime.Now.ToString(Game.Settings.Server.TimestampFormat);
	channel.Writer.WriteLine("[{0}] {1}", timestamp, format.F(args));
}
```
to improve readability and reduce redundancy.

Reply:
Done.